src/test/scala/com/microsoft/hyperspace/index/HybridScanTest.scala

```diff
@@ -291,4 +340,339 @@ class HybridScanTest extends QueryTest with HyperspaceSuite {
       }
     }
   }
+
+  test(
+    "Delete-only: filter index & parquet format, " +
+      "Hybrid Scan for delete support doesn't work without linage column") {
+    val df = spark.read.parquet(sampleParquetDataLocationDelete2)
+    def filterQuery: DataFrame =
+      df.filter(df("clicks") <= 2000).select(df("query"))
+    val baseQuery = filterQuery
+
+    withSQLConf(
+      "spark.hyperspace.index.hybridscan.enabled" -> "true",
+      "spark.hyperspace.index.hybridscan.delete.enabled" -> "false") {
+      val filter = filterQuery
+      assert(baseQuery.queryExecution.optimizedPlan.equals(filter.queryExecution.optimizedPlan))
+    }
+
+    withSQLConf(
+      "spark.hyperspace.index.hybridscan.enabled" -> "true",
+      "spark.hyperspace.index.hybridscan.delete.enabled" -> "true") {
+      val filter = filterQuery
+      assert(baseQuery.queryExecution.optimizedPlan.equals(filter.queryExecution.optimizedPlan))
+    }
+  }
+
+  test(
+    "Delete-only: filter index & parquet, json format, " +
+      "index relation should have additional filter for deleted files") {
+
+    Seq(
+      (sampleParquetDataLocationDelete, "index2", "parquet"),
+      (sampleJsonDataLocationDelete, "index5", "json")) foreach {
```

Comment:
There are so many indexes created for this suite that it's very hard to follow.

Can you at least describe what `index2` and `index5` are at the beginning of test so that I don't have to go back and forth to understand these indexes?

I am also fine creating the `index5` in this test since it's not being shared anyway (unless you are testing a scenario with multiple indexes in place).

Reply:
I renamed the indexes more descriptively. Let me know if they are still unclear. Thanks!